asset/js/setup.js

```diff
@@ -19,13 +19,52 @@ function flattenModals() {
   });
 }
 
+function insertCss(cssCode) {
+  const newNode = document.createElement('style');
+  newNode.innerHTML = cssCode;
+  document.getElementsByTagName('head')[0].appendChild(newNode);
+}
+
 function setupAnchors() {
+  const headerSelector = jQuery('header');
+  const isFixed = headerSelector.filter('.header-fixed').length !== 0;
+  const headerHeight = headerSelector.height();
+  const topPadding = 1;
+  if (isFixed) {
+    jQuery('.nav-inner').css('padding-top', `calc(${headerHeight}px + ${2 * topPadding}rem)`);
+    jQuery('#content-wrapper').css('padding-top', `calc(${headerHeight}px + ${2 * topPadding}rem)`);
+  }
   jQuery('h1, h2, h3, h4, h5, h6, .header-wrapper').each((index, heading) => {
     if (heading.id) {
       jQuery(heading).on('mouseenter',
                          () => jQuery(heading).find('.fa.fa-anchor').css('visibility', 'visible'));
       jQuery(heading).on('mouseleave',
                          () => jQuery(heading).find('.fa.fa-anchor').css('visibility', 'hidden'));
+      if (isFixed) {
+        /**
+         * Fixing the top navbar would break anchor navigation,
+         * by creating empty spans above the <h> tag we can prevent
+         * the headings from being covered by the navbar.
+         */
+        const spanId = heading.id;
+        jQuery(heading).append(
+          jQuery('<span/>',
+                 {
+                   id: spanId,
+                   class: 'anchor',
+                 },
+          ),
+        );
+        jQuery(heading).removeAttr('id'); // to avoid duplicated id problem
+        const headingHeight = jQuery(heading).height();
```

Comment:
I added some comments [here](https://github.com/MarkBind/markbind/pull/982#discussion_r368459592) with some suggestions on how to avoid using the heading height, so that we can minimise the amount of CSS we are dynamically inserting. Have you taken a look at that? Perhaps it was not visible because it was in an outdated comment chain.

Reply:
Using that approach would also mean we could just update a single anchor class to have the appropriate style, instead of having to iterate through every heading to apply the styles individually like we are doing here.